include/sfx.h

```diff
@@ -1,1286 +1,145 @@
-#define NA_SE_EV_ZELDA_POWER 0x28D9
-#define NA_SE_EV_RAIN 0x28DA
-#define NA_SE_EV_IRON_DOOR_OPEN 0x28DB
-#define NA_SE_EV_IRON_DOOR_CLOSE 0x28DC
-#define NA_SE_EV_WHIRLPOOL 0x28DD
-#define NA_SE_EV_TOWER_PARTS_BROKEN 0x28DE
-#define NA_SE_EV_COW_CRY 0x28DF
-#define NA_SE_EV_METAL_BOX_BOUND 0x28E0
-#define NA_SE_EV_ELECTRIC_EXPLOSION 0x28E1
-#define NA_SE_EV_HEAVY_THROW 0x28E2
-#define NA_SE_EV_FROG_CRY_0 0x28E3
-#define NA_SE_EV_FROG_CRY_1 0x28E4
-#define NA_SE_EV_COW_CRY_LV 0x28E5
-#define NA_SE_EV_RONRON_DOOR_CLOSE 0x28E6
-#define NA_SE_EV_BUTTERFRY_TO_FAIRY 0x28E7
-#define NA_SE_EV_FIVE_COUNT_LUPY 0x28E8
-#define NA_SE_EV_STONE_GROW_UP 0x28E9
-#define NA_SE_EV_STONE_LAUNCH 0x28EA
-#define NA_SE_EV_STONE_ROLLING 0x28EB
-#define NA_SE_EV_TOGE_STICK_ROLLING 0x28EC
-#define NA_SE_EV_TOWER_ENERGY 0x28ED
-#define NA_SE_EV_TOWER_BARRIER 0x28EE
-#define NA_SE_EV_CHIBI_WALK 0x28EF
-#define NA_SE_EV_KNIGHT_WALK 0x28F0
-#define NA_SE_EV_PILLAR_MOVE_STOP 0x28F1
-#define NA_SE_EV_ERUPTION_CLOUD 0x28F2
-#define NA_SE_EV_LINK_WARP_OUT_LV 0x28F3
-#define NA_SE_EV_LINK_WARP_IN 0x28F4
-#define NA_SE_EV_OCARINA_BMELO_0 0x28F5
-#define NA_SE_EV_OCARINA_BMELO_1 0x28F6
-#define NA_SE_EV_EXPLOSION_FOR_RENZOKU 0x28F7
+typedef enum {
+    /* 0 */ BANK_PLAYER,
+    /* 1 */ BANK_ITEM,
+    /* 2 */ BANK_ENV,
+    /* 3 */ BANK_ENEMY,
+    /* 4 */ BANK_SYSTEM,
+    /* 5 */ BANK_OCARINA,
+    /* 6 */ BANK_VOICE
+} SoundBankTypes;
+
+typedef enum {
+    /* 0 */ SFX_STATE_EMPTY,
+    /* 1 */ SFX_STATE_QUEUED,
+    /* 2 */ SFX_STATE_READY,
+    /* 3 */ SFX_STATE_PLAYING_REFRESH,
+    /* 4 */ SFX_STATE_PLAYING_1,
+    /* 5 */ SFX_STATE_PLAYING_2
+} SfxState;
+
+typedef struct {
+    /* 0x00 */ f32*     posX;
+    /* 0x04 */ f32*     posY;
+    /* 0x08 */ f32*     posZ;
+    /* 0x0C */ u8       token;
+    /* 0x10 */ f32*     freqScale;
+    /* 0x14 */ f32*     vol;
+    /* 0x18 */ s8*      reverbAdd;
+    /* 0x1C */ f32      dist;
+    /* 0x20 */ u32      priority; // lower is more prioritized
+    /* 0x24 */ u8       sfxImportance;
+    /* 0x26 */ u16      sfxParams;
+    /* 0x28 */ u16      sfxId;
+    /* 0x2A */ u8       state; // uses SfxState enum
+    /* 0x2B */ u8       freshness;
+    /* 0x2C */ u8       prev;
+    /* 0x2D */ u8       next;
+    /* 0x2E */ u8       channelIdx;
+    /* 0x2F */ u8       unk_2F;
+} SoundBankEntry; // size = 0x30
+
+/*
+ * SFX IDs
+ *
+ * index    0000000111111111    observed in audio code
+ * & 200    0000001000000000    single bit
+ * & 400    0000010000000000    single bit
+ * & 800    0000100000000000    single bit, what we currently call SFX_FLAG
+ * & 600    0000011000000000    2 bits
+ * & A00    0000101000000000    2 bits
+ * & C00    0000110000000000    2 bits, observed in audio code
+ * & E00    0000111000000000    all 3 bits
+ * bank     1111000000000000    observed in audio code
+ */
+
+#define DEFINE_SFX(enum, _1, _2, _3, _4) enum,
+
+typedef enum {
+    NA_SE_PL_BASE = 0x7FF,
+    #include "tables/sfx_playerbank_table.h"
```

Comment:
I think we should have a subfolder dedicated to these tables, to keep things a bit more manageable.
So I suggest something like this:
```suggestion
    #include "tables/sfx/playerbank_table.h"
```